File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int left=0, maxCount=0, maxLen=0;
        int arr[26];
        for(int right=0;right<s.length();right++){
            maxCount= max(maxCount, ++arr[s[right]-'A']);
            while(right-left+1-maxCount >k){
                arr[s[left]-'A']--;
                left++;
            }
            maxLen=max(maxLen, right-left+1);
        }
        return maxLen;
    }
};

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.java
class Solution {
    public int characterReplacement(String s, int k) {
        int[] freqArr = new int[26];
        int left = 0;
        int maxWinFreq = 0 ;
        int maxLen = 0;
        for(int right = 0; right < s.length();right++){
            maxWinFreq = Math.max(maxWinFreq, ++freqArr[s.charAt(right)-'A']);
            if(right-left+1 - maxWinFreq > k){
                --freqArr[s.charAt(left)-'A'];
                left++;
            }
            maxLen = Math.max(maxLen, right-left+1);
        }
        return maxLen;
    }
}